server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeChecker.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.manager.upgrade;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.accumulo.core.fate.zookeeper.ZooReaderWriter;
+import org.apache.accumulo.manager.EventCoordinator;
+import org.apache.accumulo.server.AccumuloDataVersion;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.ZKUtil;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class UpgradeChecker {
```

Comment:
What is this for? I wasn't expecting new code, since this task was to remove old code. This should have a javadoc to explain what it's for, at the least.

Reply:
This is not so much new code as it is moving existing code to a different place.

The code to upgrade to 2.1 included the check for ZooKeeper ACLs and with the deletion of that code, this check would not be performed.  Being that it is unclear how the invalid ACLs are occurring, it seemed like something that could be checked on each upgrade and not pegged to a specific upgrade version.

Reply:
Okay, checking on upgrade seems reasonable. I think maybe just adding a javadoc comment and updating names/references to it to make it clear it's a check for ZooKeeper prerequisite for the upgrade. Right now, it feels bolted on to upgrade, rather than integrated as part of the upgrade process. Polishing it up with location/naming/comments could help make it a bit more integrated into the upgrade process itself.

Reply:
Addressed in a58a977761 - Renamed the class to PreUpgradeValidation and added javadoc